lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1159,11 +1133,21 @@ static void optimizeArithmeticNodes(Function *F) {
     Node *node = worklist.back();
     worklist.pop_back();
 
-    auto *sn = simplifyNode(node, F);
-    if (sn != node) {
-      node->getNthResult(0).replaceAllUsesOfWith(sn);
+    auto *SN = simplifyNode(node, F);
+    if (SN != node) {
+      node->getNthResult(0).replaceAllUsesOfWith(SN);
       // The simplified node could be further simplified.
-      worklist.push_back(sn);
+      worklist.push_back(SN);
+
+      // The simplified node's operands could be further simplified as well.
+      if (SN->isArithmetic()) {
+        auto *LHS = SN->getNthInput(0).getNode();
+        auto *RHS = SN->getNthInput(1).getNode();
+        if (LHS->isArithmetic())
+          worklist.push_back(LHS);
+        if (RHS->isArithmetic())
+          worklist.push_back(RHS);
+      }
```

Comment:
Is that OK to push_back LHS and RHS first, then SN?

Reply:
I'm not sure what you mean here -- the operands are pushed on after the node itself. So the operands will be processed before the node.